### PR TITLE
An implementation of TriCG

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: src/tricg.jl [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+TriCG is based on the preconditioned orthogonal tridiagonalization process
+and its relation with the preconditioned block-Lanczos process.
+
+    [ M   O ]
+    [ 0   N ]
+
+indicates the weighted norm in which residuals are measured.
+It's the Euclidean norm when `M` and `N` are identity operators.
+
+TriCG stopped when `itmax` iterations are reached or when `‖rₖ‖ ≤ atol + ‖r₀‖ * rtol`.
```
Comment (dpo) [inline src/tricg.jl:37]: stops

File: src/tricg.jl [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+`atol` is an absolute tolerance and `rtol` is a relative tolerance.
+
+Additional details can be displayed if the `verbose` mode is enable.
```
Comment (dpo) [inline src/tricg.jl:40]: enabled

File: test/test_tricg.jl [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+  # Test saddle-point systems
+  A, b, D = saddle_point()
+  m, n = size(A)
+  c = -b
+  D⁻¹ = inv(D)
+  N⁻¹ = eye(n)
+  H⁻¹ = BlockDiagonalOperator(D⁻¹, N⁻¹)
+
+  (x, y, stats) = tricg(A, b, c, τ=1.0, ν=0.0, M=D⁻¹)
```
Comment (dpo) [inline test/test_tricg.jl:12]: The docstring says that tau and nu should be nonzero. But this is a great example; it should be in the docstring!

**Comment by amontoison** [comment]
Do you have other comments @dpo ?

**Comment by dpo** [comment]
Thanks!

**Pull request merged**